### PR TITLE
MoMAGIC Ad Network - Vendor Registration

### PR DESCRIPTION
**Pull request by mmprebid**
✨ New Vendor
MoMAGIC Ad Network is an all-in-one platform that help businesses to generate revenue through Advertisements. MoMAGIC supports advertising technologies like RTB (Real Time Bidding), Header Bidding, Google Ad Manager Based deployment and simplified JS Implementations. We help provide various kinds of Ad Format support ranging from simple banner, video to instream video and interactive banner/video Ads. 

This PR adds and integrates the MoMAGIC Ad Network to the amp vendor list, using the tag amp-ad.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -53,5 +53,9 @@
       "fileMatch": ["test/visual-diff/visual-tests.jsonc"],
       "url": "./build-system/json-schemas/visual-tests.json"
     }
-  ]
+  ],
+  "editor.bracketPairColorization.enabled": true,
```
Comment (erwinmombay) [inline .vscode/settings.json:57]: @danielrozenberg could you take a look at these vscode settings change

**Comment by erwinmombay** [comment]
@mmprebid could you re-push the code to re-trigger circieCI

File: .vscode/settings.json (hunk 1 above)
Comment (danielrozenberg) [reply .vscode/settings.json:57]: I think I would prefer if we don't make this change. I personally have the same setting, but in my user profile - we should only have settings that are project-specific in this file :)

File: extensions/amp-ad/amp-ad.md [hunk 2]
```diff
@@ -548,3 +548,4 @@ See [amp-ad rules](validator-amp-ad.protoascii) in the AMP validator specificati
 -   [Yahoo Native Ads](../../ads/vendors/yahoonativeads.md)
 -   [Zen](../../ads/vendors/zen.md)
 -   [ZergNet](../../ads/vendors/zergnet.md)
+-   [Momagic](../../ads/vendors/momagic.md)
```
Comment (powerivq) [inline extensions/amp-ad/amp-ad.md:551]: Please place yours in alphabetical order.

**Review by powerivq — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 1 above)
Comment (erwinmombay) [reply .vscode/settings.json:57]: @danielrozenberg agree. @mmprebid could you revert the changes in this file. thanks

Comment (mmprebid) [reply .vscode/settings.json]: @erwinmombay, 
Thanks for the replying
As suggested, we have revert the .vscode settings to previous. Please take a look and let us know for any update.

File: .vscode/settings.json [hunk 3]
```diff
@@ -54,4 +54,4 @@
       "url": "./build-system/json-schemas/visual-tests.json"
     }
   ]
-}
+}
```
Comment (powerivq) [inline .vscode/settings.json:57]: Can you also revert this?

File: 3p/integration.js [hunk 4]
```diff
@@ -177,6 +177,7 @@ import {mgid} from '#ads/vendors/mgid';
 import {microad} from '#ads/vendors/microad';
 import {miximedia} from '#ads/vendors/miximedia';
 import {mixpo} from '#ads/vendors/mixpo';
+import {momagic} from '#ads/vendors/momagic';
```
Comment (powerivq) [inline 3p/integration.js:180]: Please revert the changes to this file. It is not necessary for new vendors.

**Review by powerivq — CHANGES_REQUESTED** [review]

**Comment by powerivq** [comment]
@mmprebid thank you! added two more comments and it should be good then.

File: 3p/integration.js (hunk 4 above)
Comment (mmprebid) [reply 3p/integration.js]: @powerivq 
We have revert the changes to the previous. Please check the same.

File: extensions/amp-ad/amp-ad.md [hunk 5]
```diff
@@ -401,6 +401,7 @@ See [amp-ad rules](validator-amp-ad.protoascii) in the AMP validator specificati
 -   [Mixpo](../../ads/vendors/mixpo.md)
 -   [Monetizer101](../../ads/vendors/monetizer101.md)
 -   [mox](../../ads/vendors/mox.md)
+-   [Momagic](../../ads/vendors/momagic.md)
```
[the remark this answers is not in the document]
Comment (mmprebid) [reply extensions/amp-ad/amp-ad.md]: @powerivq 
Thanks, we have updated the alphabetical order for Momagic. Please check

**Review by powerivq — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 3 above)
Comment (mmprebid) [reply .vscode/settings.json:57]: Hi @powerivq, we have run the given command. Please check and confirm

Comment (powerivq) [reply .vscode/settings.json:57]: @mmprebid The file is still here, so it is not right. This file should disappear from this PR.

Comment (mmprebid) [reply .vscode/settings.json:57]: @powerivq 
Hi, Do we need to remove this file completely or remove some code from this file. As this file is also exists in the ampproject and we have forked the repo. 

ampproject/amphtml/.vscode/settings.json

Please let us know.

Comment (powerivq) [reply .vscode/settings.json:57]: @mmprebid I meant that this file should remain unchanged, and not to delete this file. You are change a newline from this file right now.

Comment (mmprebid) [reply .vscode/settings.json:59]: @powerivq
We have updated the file. please check and let us know

File: .vscode/settings.json [hunk 6]
```diff
@@ -56,4 +56,4 @@
...
```
Comment (powerivq) [inline .vscode/settings.json:59]:
```suggestion
}

```

File: .vscode/settings.json (hunk 3 above)
Comment (powerivq) [reply .vscode/settings.json:59]: @mmprebid It still is not working, as I still see this file on here. Could you try accepting my suggested change below?

**Review by powerivq — CHANGES_REQUESTED** [review]

**Review by powerivq — APPROVED** [review]

File: .vscode/settings.json (hunk 3 above)
Comment (mmprebid) [reply .vscode/settings.json]: @powerivq 
We have done syncing with latest updates and we have also compared the files from ampproject/amphtml. Now, Please check and let us know.

If it is not updated then please ignore this file **/.vscode/settings.json** during commit or revert the old code as we are not getting any mismatch on our repository.

**Comment by powerivq** [comment]
@mmprebid Thanks. Looks good now, but unfortunately (I have tried a few times), the build is not running. I do not know if it is because of the branch name `main` or that because I committed something into it. Can you create another branch and open another PR with the same code, and I will approve it there?

**Comment by mmprebid** [comment]
@powerivq 
We have created the new PR - #39956 as suggested. Please check and approve accordingly

**Comment by erwinmombay** [comment]
closing this PR in favor of the new one
